### PR TITLE
chore: updated loading snippets in all app examples

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-1421/update-the-snippet-version-in-all-the-app-examples

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Enhanced tracking capabilities by introducing 'setDefaultInstanceKey' and 'consent' methods in various sample apps and SDK versions.
- **Refactor**
	- Streamlined analytics initialization by removing redundant methods for improved efficiency.
- **Chores**
	- Updated RudderStack JS SDK to version '3.0.1' in multiple examples.
	- Improved performance and compatibility by adjusting script loading logic, doctype declaration, and meta tags.

### CODE REVIEW

File: examples/reactjs/hooks/sample-app/public/index.html [hunk 1]
```diff
@@ -26,26 +26,23 @@
...
-        'getUserId',
-        'getUserTraits',
-        'getGroupId',
-        'getGroupTraits',
-        'startSession',
-        'endSession',
-        'getSessionId',
-      ];
+      window.RudderSnippetVersion = '3.0.1';
+    window.rudderanalytics = [];
```
Comment (saikumarrs) [inline examples/reactjs/hooks/sample-app/public/index.html:30]: The formatting is a bit off.

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Need to correct the formatting of the snippet in some example apps.

**Comment by saikumarrs** [comment]
@MoumitaM, can you revert the "chore" inclusion change we made to the release scripts and add a comment in the "publish new release" workflow to keep using the GitHub runners?

File: examples/angular/sample-app/src/index.html [hunk 2]
```diff
@@ -7,25 +7,22 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
   <script>
+    window.RudderSnippetVersion = '3.0.1';
```
Comment (saikumarrs) [inline examples/angular/sample-app/src/index.html:10]: @MoumitaM, can you revert the "chore" inclusion change we made to the release scripts and add a comment in the "publish new release" workflow to keep using the GitHub runners?

File: examples/reactjs/hooks/sample-app/public/index.html (hunk 1 above)
Comment (MoumitaM) [reply examples/reactjs/hooks/sample-app/public/index.html:30]: prettier is not working for html files, I will check

Comment (MoumitaM) [reply examples/reactjs/hooks/sample-app/public/index.html]: addressed

File: examples/angular/sample-app/src/index.html (hunk 2 above)
Comment (MoumitaM) [reply examples/angular/sample-app/src/index.html:10]: addressed

**Review by saikumarrs — APPROVED** [review]

File: examples/v3-legacy/index.html [hunk 3]
```diff
@@ -22,11 +22,10 @@
...
         'reset',
-        'getAnonymousId',
         'setAnonymousId',
         'startSession',
         'endSession',
-        'getSessionId',
+        'consent'
       ];
       for (var i = 0; i < methods.length; i++) {
         var method = methods[i];
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply examples/v3-legacy/index.html:31]: getAnonymousId method is still supported but there is no need to buffer it. Hence we removed it from the list.

File: examples/v3-minimum-plugins/index.html [hunk 4]
```diff
@@ -22,11 +22,10 @@
...
         'reset',
-        'getAnonymousId',
         'setAnonymousId',
         'startSession',
         'endSession',
-        'getSessionId',
+        'consent'
       ];
       for (var i = 0; i < methods.length; i++) {
         var method = methods[i];
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply examples/v3-minimum-plugins/index.html:31]: getAnonymousId method is still supported but there is no need to buffer it. Hence we removed it from the list.

File: examples/v3-legacy-minimum-plugins/index.html [hunk 5]
```diff
@@ -22,11 +22,10 @@
...
         'reset',
-        'getAnonymousId',
         'setAnonymousId',
         'startSession',
         'endSession',
-        'getSessionId',
+        'consent'
       ];
       for (var i = 0; i < methods.length; i++) {
         var method = methods[i];
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply examples/v3-legacy-minimum-plugins/index.html:31]: 'setAnonymousId' method should be present

File: examples/v3-beacon/index.html [hunk 6]
```diff
@@ -22,11 +22,10 @@
...
         'reset',
-        'getAnonymousId',
         'setAnonymousId',
         'startSession',
         'endSession',
-        'getSessionId',
+        'consent'
       ];
       for (var i = 0; i < methods.length; i++) {
         var method = methods[i];
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply examples/v3-beacon/index.html:31]: 'setAnonymousId' method should be present

**Pull request merged**